src/session/session_dhandle.c

```diff
@@ -463,14 +447,23 @@ __session_get_dhandle(
 		return (0);
 	}
 
+	/* Sweep the handle list to remove any dead handles. */
+	WT_RET(__session_dhandle_sweep(session));
```

Comment:
My concern with this change is that we could kick the handle we are about to access out of cache if the sweep happens before the find. That won't break anything, but it will mean there is more work (and locking) in this path.